dev-client/src/hooks/gorhomHooks.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {
+  ForwardedRef,
+  RefObject,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+} from 'react';
+
+import {BottomSheetModal as GorhomBottomSheetModal} from '@gorhom/bottom-sheet';
+
+import {ModalHandle} from 'terraso-mobile-client/components/modals/Modal';
+
+/* Allows binding a GorhomBottomSheetModal to forwarded refs of our own ModalHandle type */
+export const useGorhomModalHandleRef = (
```

Comment:
This seemed like a good way to factor out the duplicated code between our three bottom sheets while preserving the simplicity of their single-purpose implementations. Am open to feedback for other approaches, though.